GitUp/Application/AuthenticationWindowController.m

```diff
@@ -0,0 +1,154 @@
+//
+//  AuthenticationWindowController.m
+//  Application
+//
+//  Created by Dmitry Lobanov on 08.10.2019.
+//
+
+#import "AuthenticationWindowController.h"
+#import <GitUpKit/GitUpKit.h>
+#import <GitUpKit/XLFacilityMacros.h>
+
+@interface AuthenticationWindowControllerModel : NSObject
+@property (nonatomic, assign) BOOL useKeychain;
+@property (nonatomic, copy) NSURL *url;
+@property (nonatomic, copy) NSString *name;
+@property (nonatomic, copy) NSString *password;
+
+@property (nonatomic, assign, readonly) BOOL isValid;
+@end
+
+@implementation AuthenticationWindowControllerModel
+- (BOOL)isValid {
+  return self.url && self.name && self.password;
+}
+- (void)unsetUseKeychain {
+  self.useKeychain = NO;
+}
+- (void)willStartTransfer {
+  self.useKeychain = YES;
+  self.url = nil;
+  self.name = nil;
+  self.password = nil;
+}
+- (void)didFinishTransferWithURL:(NSURL *)url success:(BOOL)success onResult:(void(^)(AuthenticationWindowControllerModel *model))onResult {
+  if (onResult) {
+    BOOL shouldPassModel = success && self.isValid;
+    onResult( shouldPassModel ? self : nil );
+  }
+  self.url = nil;
+  self.name = nil;
+  self.password = nil;
+}
+@end
+
+@interface AuthenticationWindowController ()
+
+// Model
+@property (nonatomic, strong) AuthenticationWindowControllerModel *model;
+
+// Outlets
+@property(nonatomic, weak) IBOutlet NSTextField* urlTextField;
+@property(nonatomic, weak) IBOutlet NSTextField* nameTextField;
+@property(nonatomic, weak) IBOutlet NSSecureTextField* passwordTextField;
+
+// First Responder
+- (NSResponder *)firstResponderWhenUsernameExists:(BOOL)usernameExists;
+- (void)makeFirstResponderWhenUsernameExists:(BOOL)usernameExists;
+
```

Comment:
These don't seem to be required in the AppDelegate version, what are these needed now?

Reply:
@lucasderraugh They set text field with empty `stringValue`.

Reply:
These still don't need to be declared in this interface. Let them be in the implementation only.